src/Mvc/Mvc.ViewFeatures/src/RazorComponents/HtmlRenderer.cs

```diff
@@ -140,30 +116,57 @@ private int RenderFrames(List<string> result, ArrayRange<RenderTreeFrame> frames
         }
 
         private int RenderChildComponent(
-            List<string> result,
+            HtmlRenderingContext context,
             ArrayRange<RenderTreeFrame> frames,
             int position)
         {
             ref var frame = ref frames.Array[position];
             var childFrames = GetCurrentRenderTreeFrames(frame.ComponentId);
-            RenderFrames(result, childFrames, 0, childFrames.Count);
+            RenderFrames(context, childFrames, 0, childFrames.Count);
             return position + frame.ComponentSubtreeLength;
         }
 
         private int RenderElement(
-            List<string> result,
+            HtmlRenderingContext context,
             ArrayRange<RenderTreeFrame> frames,
             int position)
         {
             ref var frame = ref frames.Array[position];
+            var result = context.Result;
             result.Add("<");
             result.Add(frame.ElementName);
-            var afterAttributes = RenderAttributes(result, frames, position + 1, frame.ElementSubtreeLength - 1);
+            var afterAttributes = RenderAttributes(context, frames, position + 1, frame.ElementSubtreeLength - 1, out var capturedValueAttribute);
+
+            // When we see an <option> as a descendant of a <select>, and the option's "value" attribute matches the
+            // "value" attribute on the <select>, then we auto-add the "selected" attribute to that option. This is
+            // a way of converting Blazor's select binding feature to regular static HTML.
+            if (context.ClosestSelectValueAsString != null
+                && string.Equals(frame.ElementName, "option", StringComparison.OrdinalIgnoreCase)
+                && string.Equals(capturedValueAttribute, context.ClosestSelectValueAsString, StringComparison.Ordinal))
+            {
+                result.Add(" selected");
+            }
+
             var remainingElements = frame.ElementSubtreeLength + position - afterAttributes;
             if (remainingElements > 0)
             {
                 result.Add(">");
-                var afterElement = RenderChildren(result, frames, afterAttributes, remainingElements);
+
+                var isSelect = string.Equals(frame.ElementName, "select", StringComparison.OrdinalIgnoreCase);
+                if (isSelect)
+                {
+                    context.ClosestSelectValueAsString = capturedValueAttribute;
+                }
+
+                var afterElement = RenderChildren(context, frames, afterAttributes, remainingElements);
+
+                if (isSelect)
+                {
+                    // There's no concept of nested <select> elements, so as soon as we're exiting one of them,
+                    // we can safely say there is no longer any value for this
+                    context.ClosestSelectValueAsString = null;
```

Comment:
Should we try and detect this case and throw in that scenario? I'm concerned between the difference between prerender and non-prerender here.

Reply:
I don't think there's any difference between prerender and non-prerender in any valid scenario. It's never valid to nest `<select>` elements.